### PR TITLE
Improve crypto log to debug UISI

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Update crypto logs to better help debug UISIs

### CODE REVIEW

**Review by ouchadam — APPROVED** [review]

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/actions/EnsureOlmSessionsForDevicesAction.kt [hunk 1]
```diff
@@ -36,22 +37,31 @@ internal class EnsureOlmSessionsForDevicesAction @Inject constructor(
...
         val results = MXUsersDevicesMap<MXOlmSessionResult>()
 
-        for ((userId, deviceInfos) in devicesByUser) {
-            for (deviceInfo in deviceInfos) {
+        for ((userId, deviceInfo) in devicesByUser) {
+            for (deviceInfo in deviceInfo) {
                 val deviceId = deviceInfo.deviceId
                 val key = deviceInfo.identityKey()
+                if (key == null) {
+                    Timber.w("## CRYPTO | Ignoring device (${deviceInfo.userId}|$deviceId) without identity key")
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/actions/EnsureOlmSessionsForDevicesAction.kt:45]: use a Timber tag in this file too?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt [hunk 2]
```diff
@@ -275,7 +285,7 @@ internal class MXMegolmEncryption(
                 gossipingEventBuffer.add(
                         Event(
                                 type = EventType.ROOM_KEY,
-                                senderId = this.userId,
+                                senderId = this.myUserId,
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt:288]: I think `this.` is not necessary anymore here

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/CryptoSyncHandler.kt [hunk 3]
```diff
@@ -32,6 +33,8 @@ import org.matrix.android.sdk.internal.session.initsync.ProgressReporter
 import timber.log.Timber
 import javax.inject.Inject
 
+private val loggerTag = LoggerTag("MXMegolmEncryption", LoggerTag.CRYPTO)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/CryptoSyncHandler.kt:36]: `MXMegolmEncryption` ? It's maybe wanted.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt (hunk 2 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt]: done

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/CryptoSyncHandler.kt (hunk 3 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/CryptoSyncHandler.kt]: good catch (bad copy paste)

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/actions/EnsureOlmSessionsForDevicesAction.kt (hunk 1 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/actions/EnsureOlmSessionsForDevicesAction.kt:48]: Done

**Comment by bmarty** [comment]
Thanks for the update.
Can you fix this compilation warning too?

> /home/runner/work/element-android/element-android/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/actions/EnsureOlmSessionsForDevicesAction.kt: (44, 18): Name shadowed: deviceInfo

**Review by bmarty — APPROVED** [review]
Thanks for the update

**Pull request merged**